frontend/src/metabase/query_builder/components/dataref/ModelPane/ModelPane.jsx

```diff
@@ -0,0 +1,84 @@
+import React from "react";
+import { connect } from "react-redux";
+import { t, jt } from "ttag";
+import { createSelector } from "reselect";
+import PropTypes from "prop-types";
+
+import { getMetadata } from "metabase/selectors/metadata";
+import MetabaseSettings from "metabase/lib/settings";
+import { formatDateTimeWithUnit } from "metabase/lib/formatting";
+import {
+  Description,
+  EmptyDescription,
+} from "metabase/components/MetadataInfo/MetadataInfo.styled";
+import Question from "metabase-lib/lib/Question";
+import FieldList from "../FieldList";
+import {
+  ModelPaneDetail,
+  ModelPaneDetailLink,
+  ModelPaneDetailText,
+  ModelPaneIcon,
+  ModelPaneDescription,
+} from "./ModelPane.styled";
+
+// This formats a timestamp as a date using any custom formatting options.
+function formatDate(value) {
+  const options = MetabaseSettings.get("custom-formatting")["type/Temporal"];
+  return formatDateTimeWithUnit(value, "day", options);
+}
+
+const getQuestion = createSelector(
```

Comment:
This selector won't work because it will be called with different cards, invalidating cache. Could you move this selector to other query builder selectors and use something like `createCachedSelector` with caching by card id to solve this?